enhancements/samples/removed-bootstrapping.md

```diff
@@ -0,0 +1,188 @@
+been identified in Jira:
+
+- regular expressions for further manipulation of the image reference in the imagestream
+- modes where failed imagestreams are automatically added to the skip list (where the administrator then
+automatically can have them added to the management set by fixing the import by mirroring the images, fixing
+the registry in question, etc.)
+- moving the samples operator out of the payload / CVO and making it an OLM operator, where the cluster
+administrator controls which imagestreams are installed when the provision the OLM operator
+- aggregating network topology details like disconnected/mirrored via our cluster config API (which was previously 
+discussed and ultimately abandoned).
+
+Bootstrapping as removed is considered the meets minimum, least impacting change to the samples operator 
+that addresses the scenarios where out of the box imagestreams pointing to registry.redhat.io will fail.
+
+## Proposal
+
+With startup today, the samples operator bootstraps as `Managed`.  That boostrapping can be made conditional,
+and instead come up as `Removed` when the samples operator sees
+
+- Over a 3 minute window during initial install, the samples operator will attempt to create a TCP connection 
+to `registry.redhat.io` and if one never succeeds, will bootstrap as `Removed`, with no samples installed.
+- The clusteroperator `Reason` fields note the inability to access the terms based registry.
+
+The cluster administrator is then free to investigate why the inability to access `registry.redhat.com` exists
+and act accordingly. Options could include:
+
+- Samples are left in `Removed`
+- Images related to samples imagestreams are mirrored into an accessible registry, and then a) `samplesRegistry` is
+updated to that accessible registry in the sample operator's config object, b) Imagestreams not mirrored are added
+to the config's `skippedImagestreams` list, c) the operator is switched to `Managed`
+- Whatever proxy or underlying network condition that prevents access to `registry.redhat.io` is addressed and then
+the samples operator is switched back to `Managed`
+
+### Risks and Mitigations
+
+1) Misidentifying a cluster as tolerating `Managed` out of the box (i.e. failing to catch a case we are trying
+to prevent with this enhancement)
+2) Bootstrapping as `Removed` for a cluster where imagestreams are importable (i.e. breaking something that works today)
+
+Also note, a fair amount of tests in `e2e-*-builds`, `e2e-aws-image-ecosystem`, `e2e-aws-jenkins`, and the subset of 
+conformance tests in e2e-aws that leverage OpenShift Builds depend on the samples as a starting point.
+
+For e2e environments for IPv6 and disconnected / restricted network, those tests will either have to be filtered 
+out of the ginkgo query or reworked to either create their own imagestreams / templates or only leverage local
+resources that do not require network access (for example, binary builds).
```

Comment:
binary builds isn't really the right alternative here.  binary builds are a way to supply source code, but this is about supplying the builder images (in the absence of valid imagestreams).

Reply:
changing `reworked to....` to

`reworked so that any needed builder/deployment images typically retrieved from registry.redhat.io are either created or preloaded in the test cluster prior to their use in any extended test cases.`